### ISSUE TITLE
VLC video source in OBS cannot switch to another soundtrack

### ISSUE THREAD

**Issue by tomato013** [obsproject/obs-studio#8346]
### Operating System Info

Windows 11

### OBS Studio Version

Other

### OBS Studio Version (Other)

28.1.2

### OBS Studio Log URL

https://obsproject.com/logs/OpQ7elv45Kr5U3nX

### Expected Behavior

Sound can be correctly output when switching to soundtrack 2.

### Current Behavior

Sound only outputs when the soundtrack setting equals 1, when it is set to 2, no sound output.

### Steps to Reproduce

1. Add an mp4 video file, which has 2 soundtracks, to the VLC video source in OBS.

2. Sound only outputs when the soundtrack setting equals 1, when it is set to 2, no sound output.

## ScreenShots
![图片](https://user-images.githubusercontent.com/74494980/221365728-50c2484d-b998-48ad-92a8-14b0a75f87a3.png)

Video File Properties
![图片](https://user-images.githubusercontent.com/74494980/221365750-ca7b3f22-46a3-4b9c-a828-6b024a052b63.png)

**Comment by Fenrirthviti** [comment]
Are you certain that the track is actually track 2? I often see arbitrary numbers in video files (particularly anime), such as track 1, then track 101, or track 11, sometimes 20, etc. The track selection in OBS will request that exact track number, not just "the second track" in a list. We don't enumerate.

**Comment by RytoEX** [comment]
Please file bugs against the latest OBS stable release or the current git commit. Please retest in OBS Studio 29.0.2 (or newer) and attach an accompanying log file.

Is this not a duplicate of #5045?

**Comment by zapf2000** [comment]
Yes, switching to a different sound layer than "default" doesn't work. I only have 1 sound layer, even there are 4 different ones. In VLC directly it works fine. Displays all sound layers corretly and when i choose the one i like, it works. But not when i run the OBS plugin. I run Ubuntu Studio 22.04, vlc 3.0.16 and OBS 29.0.2 from https://ppa.launchpadcontent.net/obsproject/obs-studio-unstable/ubuntu/ jammy main

**Comment by tomato013** [comment]
Have updated to the newest version (29.0.2), the bug still exists.

Log file: https://obsproject.com/logs/iyLT78G1oDBohfXI

Reproduce presentation:

**Comment by tomato013** [comment]
Have updated to the newest version (29.0.2), the bug still exists.

Log file: https://obsproject.com/logs/iyLT78G1oDBohfXI

Reproduce presentation: https://youtu.be/LitQOVixqTA

**Comment by tomato013** [comment]
I have uploaded the video file to Google Drive for you to test it.

https://drive.google.com/file/d/17SD9X_NRHsBDVG2bvBB_ohxSV6_zhgel/view?usp=share_link

**Comment by tomato013** [comment]
@RytoEX Hello, any progress?
